### DISCUSSION TITLE
which license model is valid?

### DISCUSSION THREAD

**Discussion by mltsg** [soketi/pws#327] [Q&A]
there are three different specifications. 
we need to know which model is correct. 

-  MIT: [readme](https://github.com/soketi/soketi#readme)
-  APL: [master](https://github.com/soketi/soketi/blob/master/LICENSE)
- GNU: [tag 0.30.1](https://github.com/soketi/soketi/blob/0.30.1/LICENSE)

please clarify.
kind regards from vienna

**Comment by rennokki** [comment]
Starting with 0.30.2 we changed to [AGPL v3](https://github.com/soketi/soketi/commit/8e1bdbd9d47d99369112f7a9815886548822cb7d). I'll change the readme to reflect this. 😁

**Reply by mltsg** [reply]
thank you for your quick reply.

can i ask you an additional question about what exactly (AGPL v3) means for our company? 
I am not a lawyer and find it a little difficult to understand the license exactly...

our scenario:
we have developed a software which we sell to customers (medical sector). we want to use soketi as a pusher alternative.
we do not offer open source software.
are we allowed to use soketi or not?
and if yes, what do we have to do? 
if no, is there any consideration that we may use soketi in the future?

btw: i <3 soketi

**Reply by rennokki** [reply]
What you can do:

- Commercial use
- Modification
- Distribution
- Patent use
- Private use

But you have to:

- License and copyright notice (keep the LICENSE file and distribute it)
- State changes (in case you modify this project)
- Disclose source (not really required for this use case, unless you're modifying it)
- Network use is distribution (Users who interact with the software via the network are given the right to receive a copy of the corresponding source code.)
- Same license (use the same AGPLv3 license when distributing)

You can use soketi for internal use or distribute it with other software, but you can't modify it or sell it as SaaS. Prior written permission is required for such a thing.

**Reply by mltsg** [reply]
thank you for the explanation.

**Reply by rennokki** [reply]
I mean, I followed some kind of Elasticsearch license: everyone is free to use under their own terms for private use, just license it the same and you can host it yourself, but you can't sell it as SaaS. Elasticsearch and Mongo do the same, so it was prior for me to have it kinda the same, as anyone would be able to take it and sell it further, while I and the contributors are here trying to close the gap for Pusher-like services for free.

Feel free to share it further in products, I really appreciate that. 👍
